python/GafferUI/GraphBookmarksUI.py

```diff
@@ -127,6 +127,42 @@ def popupFindBookmarkMenu( editor ) :
 
 	__findBookmark( editor )
 
+def setNumericBookmark( editor, numericBookmark ) :
+
+	if not isinstance( editor, GafferUI.GraphEditor ) :
+		return False
+
+	selection = editor.scriptNode().selection()
+
+	if len( selection ) == 1 :
+		node = selection[0]
+		with Gaffer.UndoScope( node.scriptNode() ) :
+			Gaffer.MetadataAlgo.setNumericBookmark( node, numericBookmark )
+			editor.graphGadget()._requestRender()  # \todo: this internal detail shouldn't be exposed here
```

Comment:
Instead of this, StandardNodeGadget should be requesting a redraw when the metadata changes.